lib/shoryuken/manager.rb

```diff
@@ -98,7 +98,8 @@ def assign(queue, sqs_msg)
     end
 
     def dispatch_batch(queue)
-      batch = @fetcher.fetch(queue, BATCH_LIMIT)
+      return if (batch = @fetcher.fetch(queue, BATCH_LIMIT)).none?
```

Comment:
@dotpyfe ☝️ was the issue, it was trying to assign/process/delete all that stuff for empty batches.